### ISSUE TITLE
'terraform init' errors with "Blocks of type "terraform" are not expected here." When utilizing backend.tf file. Worked in 12.29

### ISSUE THREAD

**Issue by CroweDadams** [hashicorp/terraform#26301]
### Terraform Version

```
Terraform v0.13.3 on Linux (also verified on MacOS command line)
```

### Terraform Configuration Files

```terraform
Standard configuration
```

### Expected Behavior
terraform should init and accept the backend.tf

### Actual Behavior
Terraform errors out with the following text. 
---
Initializing the backend...

 
Error: Unsupported block type

 
  on backend.tf line 1:
   1: terraform {

 
**Blocks of type "terraform" are not expected here.**
---

### Steps to Reproduce

Please list the full steps required to reproduce the issue, for example:
1. create a backend.tf
2. `terraform init -backend-config="backend.tf" -backend-config="key=[[some_key]].tfstate" -input=false -reconfigure`

### Additional Context

backend.tf contents
---
```terraform
terraform {
  backend "azurerm" {
    resource_group_name  = "[[resource_group_name"
    storage_account_name = "[[storage_account_name]]"
    container_name       = "[[somecontainer]]"
    access_key		 = "[[removed]]"
    key			 = "[[some-key]]"
  }
}
```

**Comment by alisdair** [comment]
Hi @CroweDadams, thanks for reporting this.

The backend configuration file should not contain a `terraform` block. Instead, it should contain the contents of your backend block. Like this:

```hcl
resource_group_name  = "[[resource_group_name]]"
storage_account_name = "[[storage_account_name]]"
container_name       = "[[somecontainer]]"
access_key           = "[[removed]]"
key                  = "[[some-key]]"
```

Hope this helps! I'm going to close this issue as Terraform is working as designed here. However, if you could let us know of any documentation which led you to put the entire `terraform` block in this file, that would be appreciated!

**Comment by CroweDadams** [comment]
Thank you, I'll go ahead and try that, though it wasn't working last time. I'll try again. The Documentation that led me to using this was found here:

https://www.terraform.io/docs/backends/config.html


Additionally, that backend.tf worked in version 12.29, it's only when upgrading to 13.3 that it stopped working.

**Comment by CroweDadams** [comment]
Tried this and I got an error for each key. 

` terraform init -backend-config="key=[[removed]].tfstate" -backend-config="backend.tf" -input=false -reconfigure
There are some problems with the configuration, described below.

The Terraform configuration must be valid before initialization so that
Terraform can determine which modules and providers need to be installed.

Error: Unsupported argument

  on backend.tf line 1:
   1: resource_group_name  = "[[removed]]"

An argument named "resource_group_name" is not expected here.


Error: Unsupported argument

  on backend.tf line 2:
   2: storage_account_name = "[[removed]]"

An argument named "storage_account_name" is not expected here.


Error: Unsupported argument

  on backend.tf line 3:
   3: container_name       = "[[removed]]"

An argument named "container_name" is not expected here.


Error: Unsupported argument

  on backend.tf line 4:
   4: access_key           = "[[removed]]"

An argument named "access_key" is not expected here.


Error: Unsupported argument

  on backend.tf line 5:
   5: key	             = "[[removed]]"

An argument named "key" is not expected here.
`

**Comment by alisdair** [comment]
> The Documentation that led me to using this was found here 

Thanks! We recently updated those docs, and you might not have noticed [this new section on backend configuration override files](https://www.terraform.io/docs/backends/config.html#backend-configuration-file):

> A backend configuration file has the contents of the backend block as top-level attributes, without the need to wrap it in another `terraform` or `backend` block:
> 
> ```hcl
> address = "demo.consul.io"
> path    = "example_app/terraform_state"
> scheme  = "https"
> ```

On your follow-up:

> Tried this and I got an error for each key.

Ah, I see what's going on. This is caused by having your backend configuration override file named `backend.tf`—or alternatively, attempting to use the Terraform file containing your backend configuration as an override file.

Your original `backend.tf` file is completely valid Terraform configuration, and if that's the only definition for your backend, the original code was correct. However, in that case, your original command is incorrect—you need to remove the `-backend-config="backend.tf"` argument.

The reason you saw this working in Terraform 0.12.29 is because Terraform ignored that argument. You don't need to provide it unless you want to override the backend configuration in your main `.tf` files.

So, in summary, I think you want to:

- Undo the changes you just made to `backend.tf` (sorry! 😅)
- Remove the `-backend-config="backend.tf"` argument from your `init` command

Hope that works!

**Comment by CroweDadams** [comment]
That works for me, and no worries! This is what engineers do, we figure stuff out and make solutions! Thank you so much for your quick and prompt responses! 

I'll give this a shot!
